File: QytetetRuby/lib/sorpresa.rb
#encoding: utf-8 

module ModeloQytetet

class Sorpresa
  
  attr_accessor :texto
  attr_accessor :valor
  attr_accessor :tipo
  
  def initialize(texto, valor, tipo)
    @texto=texto
    @valor=valor
    @tipo=tipo    
  end
  
  def to_s()
    " \n Texto: #{@texto} \n Valor: #{@valor} \n Tipo: #{@tipo}"
  end

end

end

File: QytetetJava/src/modeloqytetet/Sorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author albertorodriguez
 */
public class Sorpresa {
    
    private String texto;
    private TipoSorpresa tipo;
    private int valor;
    
    
    //----------------------------CONSTRUCTOR--------------------------------    
    public Sorpresa(String texto, int valor, TipoSorpresa tipo){
        this.texto=texto;
        this.valor=valor;
        this.tipo=tipo;
    }


    //-------------------------CONSULTORES----------------------------------

    String getTexto(){
        return this.texto;
    }
    
    TipoSorpresa getTipo(){
        return this.tipo;
    }
    
    int getValor(){
        return this.valor;
    }
    
    
    //-------------------------TOSTRING---------------------------------- 
    @Override
        public String toString() {
             return "Sorpresa{" + "texto=" + texto + ", valor=" +
                   Integer.toString(valor) + ", tipo=" + tipo + "}";
    }
    
    
    
    
}
